custom-plugin-rules/markdown-validator/rule-validate-markdown.js

```diff
@@ -0,0 +1,77 @@
+module.exports = ValidateMarkdown
+const markdownlint = require("markdownlint");
+const config = {
+  // the list is here https://github.com/DavidAnson/markdownlint#rules--aliases
+  MD013: {line_length: 120},
+  MD041: false, // first line should be h1
+  MD047: false, // should end with newline
+}
+
+function checkString(description, ctx) {
+  let options = {
+    "strings": {
+      "desc": description
+    },
+    "config": config
+  };
+  markdownlint(options, function callback(err, result) {
+    if (!err) {
+      // if there's no problem do nothing
+      if (result.desc.length) { // desc is the key in the options.strings object
+        let lines = description.split("\n");
+
+        result.desc.forEach((desc) => {
+          message = desc.ruleDescription;
+          // add line number context for longer entries
+          if (desc.lineNumber > 1 ) {
+            message = message + " (near: " + lines[desc.lineNumber].substring(0,20) + "... )";
+          }
+
+          ctx.report({
+            message: message,
+            location: ctx.location.child('description'),
+          });
+        });
+
+      }
+    }
+  });
+}
+
+function ValidateMarkdown() {
+  console.log("OpenAPI Markdown: validate");
+  return {
+   Info: {
```

Comment:
Why not to simply check any node that contains `description`? 
Also, does it cover the case when a description contains a reference to a file?

Reply:
Why not check any node: I couldn't work out if that was safe to abstract, or if it might start checking things that might not be markdown.

When a description is a reference to a file: no, because that is not valid in OpenAPI and I have some other ideas for that. Also if you use files for your descriptions, you can validate using external tools, you don't need this openapi-aware wrapper plugin.

Reply:
> things that might not be markdown.

I'd expect any field with the name `description` to eventually converge to Markdown 🙂 
But you're right, maybe it's just better to be as much explicit as possible since this is supposed to be an example.

Reply:
Could it trip up on something else named `description` though? Like a parameter or a property? One of the big features of Redocly's linting is that it does know what things are. Or could I turn this inside out and do some sort of "any" and then check for types (and would that be more or less readable than what I actually did here?)

Reply:
You're right, I was simply missing that descriptions can contain schemas with arbitrary properties.